app/controllers/admin/orders_controller.rb

```diff
@@ -1,10 +1,87 @@
 class Admin::OrdersController < AdminController
+  before_action :load_order, except: :index
+
   def index
     @status = params[:status]
     @orders = Order.newest_first
-    @orders = @orders.status(@status) if Order.statuses.keys
-                                              .include? params[:status]
+    if Order.statuses.keys.include? params[:status]
+      @orders = @orders.status(@status)
+    else
+      @status = "all"
+    end
     @orders = @orders.paginate page: params[:page],
                                per_page: Settings.order.per_page
   end
+
+  def forward
+    if is_order_valid_for_forwarding? @order
+      change_to_next_status @order
+    else
+      flash[:warning] = t "order.messages.not_for_forwarding"
+    end
+    redirect_to admin_orders_path(status: params[:status])
+  end
+
+  def backward
+    if is_order_valid_for_backwarding? @order
+      change_to_previous_status @order
+    else
+      flash[:warning] = t "order.messages.not_for_backwarding"
+    end
+    redirect_to admin_orders_path(status: params[:status])
+  end
+
+  def reject
+    if is_order_valid_for_rejecting? @orders
+      save_canceled_rejected_order :rejected, @order
+    else
+      flash[:warning] = t "order.messages.not_for_rejecting"
+    end
+    redirect_to admin_orders_path(status: params[:status])
+  end
```

Comment:
Bạn thử áp dụng kỹ thuật meta programing vào 3 method này xem. Mình thấy 3 method này giống giống nhau chỉ khác một vài chỗ

Reply:
Ừm nhỉ, mình cũng không nghĩ đến việc này. Nhưng mình sợ áp dụng meta_programing vào ở đây hơi khó đọc thôi, vì bản thân mình thấy các hàm này sẵn đã hơi khó đọc.  
Cộng thêm các method không giống nhau hoàn toàn, nên xử lí sẽ hơi cồng kềnh 1 chút. Như method reject thì có method con có 2 argument. Trong khi method con nằm trong backward và forward có 1 argument thôi.